src/web/js/dashboard/GoogleAPI.js

```diff
@@ -93,31 +110,449 @@ class GoogleAPI {
+  All of these fields are required to execute this method. If one is missing, an error will be thrown
+  */
+  addStudent = (studentInfo) => {
+    // validation
+    for (var key in studentInfo) {
+      if (key !== "firstName" &&
+          key !== "lastName" &&
+          key !== "email"){
+        //TODO: not sure of best way to throw an error
+        return undefined;
+      }
+    }
+
+    return this.getPyretData().then((response) => {
+      //modify data
+      var data = response.result;
+
+      studentInfo.id = data.nextStudentID;
+      studentInfo.classes = [];
+
+      data.nextStudentID += 1;
+
+      //data.studentList.push(studentInfo)
+      data.studentList[studentInfo.id] = studentInfo;
+      //send data back to google
+      return this.savePyretData(data).then(() => {
+        return studentInfo;
+      });
+    });
+  }
+
+  removeStudent = (studentID) => {
+    //needs to remove student from all classes they are in
+    return this.getPyretData().then((response) => {
+      //I'm pretty sure this doesn't work, but it was apparently working for Kevin...
+      var data = response.result;
+      if (studentID in data.studentList){
+        for (var key in data.classList){
+          var index = data.classList[key].students.indexOf(studentID);
+          if (index !== undefined){
+            data.classList[key].students.splice(index, 1);
+          }
+        }
+        delete data.studentList[studentID];
+      }
+      return this.savePyretData(data);
+    });
+  }
+
+  getStudent = (studentID) => {
+    return this.getPyretData().then((response) => {
+      var data = response.result;
+      return data.studentList[studentID];
+    });
+  }
+
+  getAllStudents = () => {
+    return this.getPyretData().then((response) => {
+      var data = response.result;
+      return data.studentList;
+    });
+  }
+
+  getStudentsInClass = (classID) => {
+    return this.getClass(classID).then((classInfo) => {
+      var studentIDs = classInfo["students"];
+      return this.getAllStudents().then((studentInfo) => {
+        var courseRoster = [];
+        for (var i = 0; i < studentIDs.length; i++){
+          var studentObject = studentInfo[studentIDs[i]];
+          courseRoster.push(studentObject);
+        }
+        return courseRoster;
+      });
+    });
+  }
+
+  updateStudent = (studentID, studentInfo) => {
+    //TODO: validate the student info
+    return this.getPyretData().then((response) => {
+      var data = response.result;
+      if (studentID in data.studentList){
+        data.studentList[studentID] = studentInfo;
+      }
+      else {
+        //TODO: not sure what's the best way to throw an error here.
+        return undefined;
+      }
+      return this.savePyretData(data);
+    });
+  }
+
+  addExistingStudentToClass = (studentID, classID) => {
+    return this.getPyretData().then((response) => {
+      var data = response.result;
+
+      if (classID in data.classList && studentID in data.studentList){
```

Comment:
Made a new issue in #173. I don't think this case is within the scope of the pull request, so let's move discussion over to there. I'll add your comments, Joe.